File: NewsDaily/app/src/main/java/com/andruid/magic/newsdaily/headlines/NewsDataSourceFactory.java
package com.andruid.magic.newsdaily.headlines;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;
import androidx.paging.PageKeyedDataSource;

import com.andruid.magic.newsloader.model.News;

public class NewsDataSourceFactory extends DataSource.Factory<Integer, News> {
    private MutableLiveData<PageKeyedDataSource<Integer, News>> liveDataSource =
            new MutableLiveData<>();
    private String country, category;

    NewsDataSourceFactory(String country, String category){
        this.country = country;
        this.category = category;
    }

    @NonNull
    @Override
    public DataSource<Integer, News> create() {
        NewsDataSource dataSource = new NewsDataSource(country, category);
        liveDataSource.postValue(dataSource);
        return dataSource;
    }
}

File: NewsDailyJava/app/src/main/java/com/andruid/magic/newsdaily/headlines/NewsDataSourceFactory.kt
package com.andruid.magic.newsdaily.headlines

import androidx.paging.DataSource
import com.andruid.magic.newsloader.model.News

class NewsDataSourceFactory(val country : String, val category : String) : DataSource.Factory<Int, News>() {

    override fun create(): DataSource<Int?, News?> {
        return NewsDataSource(country, category)
    }
}